pkg/manager/mce.go

```diff
@@ -0,0 +1,373 @@
+	platformCreds.SetNamespace(clusterName)
+	platformCreds.SetName(fmt.Sprintf("%s-%s-creds", clusterName, platform))
+	if _, err := clusterSecretsClient.Create(context.TODO(), platformCreds, metav1.CreateOptions{}); err != nil {
+		return nil, fmt.Errorf("failed to create platform (%s) credentials: %v", platform, err)
+	}
+	pullSecret, err := chatBotSecretsClient.Get(context.TODO(), "mce-pull-secret", metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pull secret: %v", err)
+	}
+	// reset metadata
+	pullSecret.ObjectMeta = metav1.ObjectMeta{}
+	pullSecret.TypeMeta = metav1.TypeMeta{}
+	pullSecret.SetNamespace(clusterName)
+	pullSecret.SetName(fmt.Sprintf("%s-pull-secret", clusterName))
+	if _, err := clusterSecretsClient.Create(context.TODO(), pullSecret, metav1.CreateOptions{}); err != nil {
+		return nil, fmt.Errorf("failed to create pull secret: %v", err)
+	}
+	var baseDomain string
+	if platform == "gcp" {
+		baseDomain = fmt.Sprintf("openshift-crt-mce.%s.devcluster.openshift.com", platform)
+	} else {
+		baseDomain = fmt.Sprintf("crt-mce-%s.devcluster.openshift.com", platform)
+	}
+	if platform == "aws" && mceUserConfig.AwsBaseDomain != "" {
+		baseDomain = mceUserConfig.AwsBaseDomain
+	} else if platform == "gcp" && mceUserConfig.GcpBaseDomain != "" {
+		baseDomain = mceUserConfig.GcpBaseDomain
+	}
+	replicas := int64(3)
+	// copied from autogenerated installConfig from web interface
+	installConfig := &installer.InstallConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterName,
+		},
+		BaseDomain: baseDomain,
+		ControlPlane: &installer.MachinePool{
+			Hyperthreading: "Enabled",
+			Name:           "master",
+			Replicas:       &replicas,
+		},
+		Compute: []installer.MachinePool{{
+			Hyperthreading: "Enabled",
+			Name:           "worker",
+			Replicas:       &replicas,
+		}},
+		Networking: &installer.Networking{
+			NetworkType: "OVNKubernetes",
+			ClusterNetwork: []installer.ClusterNetworkEntry{
+				{
+					CIDR:       *ipnet.MustParseCIDR("10.128.0.0/14"),
+					HostPrefix: 23,
+				},
+			},
+			MachineNetwork: []installer.MachineNetworkEntry{
+				{
+					CIDR: *ipnet.MustParseCIDR("10.0.0.0/16"),
+				},
+			},
+			ServiceNetwork: []ipnet.IPNet{*ipnet.MustParseCIDR("172.30.0.0/16")},
+		},
+		PullSecret: "", // skip, hive will inject based on it's secrets
+	}
+	if platform == "aws" {
+		installConfig.ControlPlane.Platform = installer.MachinePoolPlatform{
+			AWS: &aws.MachinePool{
+				Zones: []string{"us-east-1a"},
+				EC2RootVolume: aws.EC2RootVolume{
+					IOPS: 2000,
+					Size: 100,
+					Type: "io1",
+				},
+				InstanceType: "m5.xlarge",
+			},
+		}
+		installConfig.Compute[0].Platform = installer.MachinePoolPlatform{
+			AWS: &aws.MachinePool{
+				Zones: []string{"us-east-1a"},
+				EC2RootVolume: aws.EC2RootVolume{
+					IOPS: 2000,
+					Size: 100,
+					Type: "io1",
+				},
+				InstanceType: "m5.xlarge",
+			},
+		}
```

Comment:
We should probably bump up to `m6a` or even `m6g` (if possible?), because they are more modern and cheaper.

Reply:
Changed to `m6a`.
`m6g` is arm64 based, so we might not want to use that as the default. Maybe it can be an option later on

Reply:
Yep.  The arm64 nodes are just slightly cheaper, that's all.  Having them as an option is a good idea.  Do you know if we might be able to utilize spot instances for worker nodes?

Reply:
It doesn't look like you can configure the cluster to use spot instances via the install-config.yaml. Our prowjobs do it by outputting the yaml of all manifests, modifying them with yq, and then passing the updated manifests to openshift-install. There's no way in MCE's web UI either.

Reply:
Yeah, I was afraid of that.  Oh well.  Hopefully, it's something MCE is working on.